lib/amazonka/src/Amazonka/Auth/ConfigFile.hs

```diff
@@ -0,0 +1,230 @@
+-- |
+-- Module      : Amazonka.Auth.ConfigFile
+-- Copyright   : (c) 2013-2021 Brendan Hay
+-- License     : Mozilla Public License, v. 2.0.
+-- Maintainer  : Brendan Hay <brendan.g.hay+amazonka@gmail.com>
+-- Stability   : provisional
+-- Portability : non-portable (GHC extensions)
+--
+-- Retrieve authentication credentials from AWS config/credentials files.
+module Amazonka.Auth.ConfigFile where
+
+import Amazonka.Auth.Container (fromContainerEnv)
+import Amazonka.Auth.Exception
+import Amazonka.Auth.InstanceProfile (fromDefaultInstanceProfile)
+import Amazonka.Auth.Keys (fromKeysEnv)
+import Amazonka.Auth.STS (fromAssumedRole, fromWebIdentity)
+import Amazonka.Data
+import Amazonka.Env (Env, Env' (..), lookupRegion)
+import Amazonka.Prelude
+import Amazonka.Types
+import qualified Control.Exception as Exception
+import Control.Exception.Lens (catching_, _IOException)
+import Data.Foldable (asum)
+import Data.HashMap.Strict as HashMap
+import qualified Data.Ini as INI
+import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
+import qualified System.Directory as Directory
+import qualified System.Environment as Environment
+
+-- | Retrieve credentials from the AWS config/credentials files, as
+-- Amazonka currently understands them:
+--
+-- * AWS recommends credentials do not live in the config file, but
+--   allows it.
+--
+-- * Sections in the config file start should either be named
+--   @[default]@ or @[profile foo]@, but unprefixed @[foo]@ is also
+--   supported.
```

Comment:
Is unprefixed format really supported? [The documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html) says 
> When naming the profile in a `config` file, include the prefix word "profile".

Reply:
I have some unprefixed `[foo]` sections in my profile where I hand-edited it, and they work.

Reply:
"Supported" usually means it has official support from the vendor. I think "happens to work" would be more accurate.

See: https://github.com/boto/botocore/blob/develop/botocore/configloader.py#L216